spyderlib/widgets/tests/test_array_builder.py

```diff
@@ -0,0 +1,204 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © 2009- The Spyder Development Team
+# Licensed under the terms of the MIT License
+#
+
+"""
+"""
+
+# Third party imports
+from qtpy.QtCore import Qt
+from pytestqt import qtbot  # analysis:ignore
+
+# Local imports
+from spyderlib.widgets.arraybuilder import NumpyArrayDialog
+
+
+class TestNumpyArrayBuilder:
+    def test_array_inline_array(self, qtbot):  # analysis:ignore
+        dlg = NumpyArrayDialog(inline=True)
+        dlg.show()
+        qtbot.addWidget(dlg)
+
+        qtbot.keyClicks(dlg.widget, '1 2 3  4 5 6')
+        qtbot.keyPress(dlg.widget, Qt.Key_Return)
+
+        expected_value = 'np.array([[1, 2, 3],\n          [4, 5, 6]])'
+        value = dlg.text()
+        assert expected_value == value
+
+    def test_array_inline_matrix(self, qtbot):  # analysis:ignore
+        dlg = NumpyArrayDialog(inline=True)
+        dlg.show()
+        qtbot.addWidget(dlg)
+
+        qtbot.keyClicks(dlg.widget, '4 5 6  7 8 9')
+        qtbot.keyPress(dlg.widget, Qt.Key_Return, modifier=Qt.ControlModifier)
+
+        expected_value = 'np.matrix([[4, 5, 6],\n           [7, 8, 9]])'
+        value = dlg.text()
+        assert expected_value == value
+
+    def test_array_inline_array_invalid(self, qtbot):  # analysis:ignore
+        dlg = NumpyArrayDialog(inline=True)
+        dlg.show()
+        qtbot.addWidget(dlg)
+
+        qtbot.keyClicks(dlg.widget, '1 2  3 4  5 6 7')
+        qtbot.keyPress(dlg.widget, Qt.Key_Return)
+        dlg.update_warning()
+
+        assert not dlg.is_valid()
+
+    def test_array_inline_1d_array(self, qtbot):  # analysis:ignore
+        dlg = NumpyArrayDialog(inline=True)
+        dlg.show()
+        qtbot.addWidget(dlg)
+
+        qtbot.keyClicks(dlg.widget, '4 5 6')
+        qtbot.keyPress(dlg.widget, Qt.Key_Return, modifier=Qt.ControlModifier)
+
+        expected_value = 'np.matrix([4, 5, 6])'
+        value = dlg.text()
+        assert expected_value == value
```

Comment:
I think that for short values there is no need to use explicit variables, as it "hides" the case being tested. In this case I prefer:
```python
        assert dlg.text() == 'np.matrix([4, 5, 6])'
```

Test files tent to be very long, if we can shorten the tests it can be a big win.